### PR TITLE
fix: keep tail spaces in heading

### PR DESCRIPTION
**Pull request by s-alves10**
### Fixed Issues
$ https://github.com/Expensify/App/issues/27445

# Tests

```
    Input: <h1>heading1</h1> in the beginning of the line
    Output: # heading1\n in the beginning of the line
```

1. Open any chat and send message
```
# heading 1
    in the beginning of the line
```
2. Switch the message just sent to edit mode
3. Verify that spaces in the second line not disappeared

# QA
1. Open any chat and send message
```
# heading 1
    in the beginning of the line
```
2. Switch the message just sent to edit mode
3. Verify that spaces in the second line not disappeared

### CODE REVIEW

**Comment by s-alves10** [comment]
@jasperhuangg 

Please take a look

**Comment by s-alves10** [comment]
@parasharrajat 

These are `htmlToMarkdown` tests for heading
https://github.com/Expensify/expensify-common/blob/cb1c6b75cb593990ed18381ba7dba8e8ceb9f16b/__tests__/ExpensiMark-Markdown-test.js#L542-L590

**Comment by s-alves10** [comment]
@parasharrajat 

`[^\S\r\n]*` means that this would match all space characters except `\r` and `\n`
But when converting markdown to html, we remove one newline and convert all other newlines to `<br />` and so there is no `\r` or `\n` in the stored html string. 

```
# heading
  next line
```
=> `<h1>heading</h1>  next line`

```
# heading

  next line
```
=> `<h1>heading</h1><br />  next line`

The original regex works fine for the latter case but doesn't work for the first case(this is just our issue) because there is no `<br />` or `\r` or `\n`

I hope this makes sense

**Comment by s-alves10** [comment]
@parasharrajat 

> My question would be why do we remove one newline after the heading tag?

I think removing one newline is reasonable because the newline in heading in markdown expresses the heading is ended. It doesn't mean newline.

**Comment by s-alves10** [comment]
@parasharrajat 

What should I do next? This PR is 8 days old without any change. I'm not sure why

**Comment by s-alves10** [comment]
@marcaaron 

Please take a look

**Comment by s-alves10** [comment]
@marcaaron 

What's the trouble with you? This PR was opened 2 weeks ago. Not sure what I can do here

**Comment by marcaaron** [comment]
Couple things. 

1. I have been waiting for @parasharrajat to review. I see he has completed his review on the 26th. That is exactly 4 business days ago.
2. We have been traveling for our company offsite in Indonesia. Not sure if we made any announcement this time around, but during this week we apply maximum focus to some internal goals. Which unfortunately can cause delays.

Last thing...

>What's the trouble with you?

Just some feedback here - but that question commonly reads as having a critical tone. I am going to assume best intentions and believe that you didn't mean it that way. It would be better to ask: "How are things looking?" or "Can I please have a status update?", "Can I help unblock you in anyway?" etc. Those are all more polite ways to check in with a reviewer.

**Review by marcaaron — APPROVED** [review]
LGTM! Thanks for your patience.

**Comment by s-alves10** [comment]
@marcaaron 

I'm sorry if I was not polite. That's not my intention

**Pull request merged**
